.goreleaser.yml

```diff
@@ -0,0 +1,68 @@
+before:
+  hooks:
+    - mkdir -p caddy-build
+    - cp cmd/caddy/main.go caddy-build/main.go
+    - cp ./go.mod caddy-build/go.mod
+    - sed -i.bkp s/github.com\/caddyserver\/caddy\/v2/caddy/g ./caddy-build/go.mod
+    # GoReleaser doesn't seem to offer {{.Tag}} at this stage, so we have to embed it into the env
+    # so we run: TAG=$(git describe --abbrev=0) goreleaser release --rm-dist --skip-publish --skip-validate
+    - go mod edit -require=github.com/caddyserver/caddy/v2@{{.Env.TAG}} ./caddy-build/go.mod
+    - git clone --depth 1 https://github.com/caddyserver/dist caddy-dist
+    - go mod download
+builds:
+- env:
+  - CGO_ENABLED=0
+  - GO111MODULE=on
+  main: main.go
+  dir: ./caddy-build
+  binary: caddy
+  goos:
+  - darwin
+  - linux
+  - windows
+  - freebsd
+  flags:
+  - -trimpath
+  ldflags:
+  - -s -w
+archives:
+- replacements:
+    darwin: Darwin
+    linux: Linux
+    windows: Windows
+    386: i386
+    amd64: x86_64
+checksum:
+  algorithm: sha256
+# signs:
+# - artifacts: all
+#   cmd: signify
+#   args: ["-S", "-s", "../seckey", "-m", "${artifact}"]
```

Comment:
Do I need to set a private key somewhere?

Reply:
https://github.com/caddyserver/caddy/pull/3173/files#diff-56c30d9224897b08c28075052d4595c5R51

Yeah in the repo secrets like we did for fuzzit.

https://github.com/goreleaser/goreleaser-action#signing

Reply:
K, thanks. I've set the signing key as a secret named `SIGNING_KEY`.

Reply:
Is it a `signify` secret key or `gpg` or something else? I put `singinfy` just for testing, but [it can be any command](https://goreleaser.com/customization/#Signing).

Here comes the paradox of choice... we need to decide on these.

Checksum algorithm:
- [ ] sha256
- [ ] sha512
- [ ] sha1
- [ ] crc32
- [ ] md5
- [ ] sha224
- [ ] sha384

Signing command:
- [ ] signify
- [ ] gpg
- [ ] gpg2
- [ ] minisign

What to sign:
- [ ] checksums
- [ ] all (executables, archives, checksums)

Reply:
It's a GPG key.

Probably sign everything with sha256 or sha512 using gpg or gpg2. How's that sound?